src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,144 @@
+        Scanner in = new Scanner(System.in);
+        boolean looper = true;
+
+        while(looper) {
+            line = in.nextLine();
+            String lineCmd = line.trim().split(" ")[0];
+            switch(lineCmd){
+            case "bye":
+                displayByeMsg();
+                looper = false;
+                break;
+            case "list":
+                cmdList(array);
+                break;
+            case "done":
+                cmdDone(array, line);
+                break;
+            case "todo":
+                cmdTodo(array, line);
+                break;
+            case "deadline":
+                cmdDeadline(array, line);
+                break;
+            case "event":
+                cmdEvent(array, line);
+                break;
+            default:
+                System.out.println("Invalid command.");
+            } //end switch statement
+        } //end while loop
+    }
+
+    /* method for event command */
+    public static void cmdEvent(Task[] array, String line) {
+        try {
+            int atPos = line.indexOf("/at");
+            String eventContent = line.trim().substring(6, atPos-1);
+            String eventAt = line.trim().split("/at ")[1];
+            Event newEvent = new Event(eventContent, eventAt);
+            array[Task.getNumTask()-1] = newEvent;
+            System.out.println("Got it. I've added this task:");
+            System.out.println("\t[E][\u2718] " + eventContent + " (at: " + eventAt + ")");
+            System.out.println("Now you have " + Task.getNumTask() + " tasks in the list.");
+        } catch (Exception e) {
+            System.out.println("Please enter a valid event.");
+        }
+    }
+
+    /* method for deadline command */
+    public static void cmdDeadline(Task[] array, String line) {
+        try {
+            int byPos = line.indexOf("/by");
+            String dlContent = line.trim().substring(9, byPos-1);
+            String dlBy = line.trim().split("/by ")[1];
+            Deadline newDeadline = new Deadline(dlContent, dlBy);
+            array[Task.getNumTask()-1] = newDeadline;
+            System.out.println("Got it. I've added this task:");
+            System.out.println("\t[D][\u2718] " + dlContent + " (by: " + dlBy + ")");
+            System.out.println("Now you have " + Task.getNumTask() + " tasks in the list.");
+        } catch (Exception e) {
+            System.out.println("Please enter a valid deadline.");
+        }
+    }
+
+    /* method for todo command */
+    public static void cmdTodo(Task[] array, String line) {
+        try {
+            String todoContent = line.trim().substring(5);
+            Todo newTodo = new Todo(todoContent);
+            array[Task.getNumTask()-1] = newTodo;
+            System.out.println("Got it. I've added this task:");
+            System.out.println("\t[T][\u2718] " + todoContent);
+            System.out.println("Now you have " + Task.getNumTask() + " tasks in the list.");
```

Comment:
Since these are repeated across all command methods, why not abstract it into a new method?